### PR TITLE
Build docker container (SOFTWARE-5154)

### PR DESCRIPTION
**Pull request by edquist**
- add Dockerfile
- add github build-container workflow

### CODE REVIEW

**Comment by edquist** [comment]
@brianhlin - how does this look?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM library/python:3.7-alpine
+
+LABEL maintainer OSG Software <help@opensciencegrid.org>
+
+COPY *.py /usr/local/bin/
+
+ENTRYPOINT ["/usr/local/bin/group_fixup.py", "--fix-all"]
```
Comment (brianhlin) [inline Dockerfile]: Let's avoid a default entrypoint and just specify the command + options in the k8s manifest https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#define-a-command-and-arguments-when-you-create-a-pod

**Comment by edquist** [comment]
Dropped

File: .github/workflows/build-container.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      run: echo "::set-output name=dtag::$(date +%Y%m%d-%H%M)"
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [make-date-tag]
+    if: startsWith(github.repository, 'opensciencegrid/')
+    strategy:
+      fail-fast: False
+      matrix:
+        repo: ['latest']
```
Comment (brianhlin) [inline .github/workflows/build-container.yml]:
```suggestion
```

A matrix of zero is a little silly

File: .github/workflows/build-container.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Generate tag list
+      id: generate-tag-list
+      env:
+        REPO: ${{ matrix.repo }}
```
Comment (brianhlin) [inline .github/workflows/build-container.yml]:
```suggestion
```

File: .github/workflows/build-container.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        TIMESTAMP: ${{ needs.make-date-tag.outputs.dtag }}
+      run: |
+        #docker_repo=${GITHUB_REPOSITORY/opensciencegrid\/docker-/opensciencegrid/}
+        docker_repo=opensciencegrid/comanage-utils
+        tag_list=()
+        for registry in hub.opensciencegrid.org docker.io; do
+          for image_tag in "$REPO" "$REPO-$TIMESTAMP"; do
```
Comment (brianhlin) [inline .github/workflows/build-container.yml]:
```suggestion
          for image_tag in "latest" "latest-$TIMESTAMP"; do
```

File: .github/workflows/build-container.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        registry: hub.opensciencegrid.org
+        username: ${{ secrets.OSG_HARBOR_ROBOT_USER }}
+        password: ${{ secrets.OSG_HARBOR_ROBOT_PASSWORD }}
+
+    - name: Build and push Docker images
+      uses: docker/build-push-action@v2.2.0
+      with:
+        context: .
+        push: true
+        tags: "${{ steps.generate-tag-list.outputs.taglist }}"
```
Comment (brianhlin) [inline .github/workflows/build-container.yml:68]: One thing that I've started doing is turning this into two steps, one for Harbor and one for Docker Hub so if Harbor's down, it doesn't blow up the whole push. You can make sure a step always runs, regardless of previous step failures with something like https://github.com/opensciencegrid/images/blob/main/.github/workflows/build-containers.yml#L168

You'll also need to fixup the taglist to separate the hub.opensciencegrid.org vs docker.io tags

**Review by brianhlin — CHANGES_REQUESTED** [review]
A request to make the pushes more robust

File: .github/workflows/build-container.yml (hunk 2 above)
Comment (edquist) [reply .github/workflows/build-container.yml]: it's a matrix of one though ...........

File: .github/workflows/build-container.yml (hunk 4 above)
Comment (edquist) [reply .github/workflows/build-container.yml]: Fine but it was one less thing to hard code with the matrix

File: .github/workflows/build-container.yml (hunk 2 above)
Comment (brianhlin) [reply .github/workflows/build-container.yml]: Ah yeah, I meant one!

Comment (edquist) [reply .github/workflows/build-container.yml]: >matrix of zero
>matrix of one

in college, for a brief revelatory period, i seemed to discover the deepest of mathematical truths:

the zero-by-zero matrix sits undetected at the deepest foundations of all mathematics.

and yet, vacuously so.

In the zero-by-zero dimension, every matrix is the Zero matrix _and_ every matrix is the One matrix, and that matrix, which is every matrix, is the Determinant.

Call it silly if you like, but mind equalled blown, i'm telling you man.

File: .github/workflows/build-container.yml (hunk 5 above)
Comment (edquist) [reply .github/workflows/build-container.yml:68]: >two steps, one for Harbor and one for Docker Hub so if Harbor's down, it doesn't blow up the whole push

@brianhlin, would we perhaps get that same behavior you are describing by just putting `docker.io` first in the registry list, before harbor?  (Assuming that Docker Hub being down is a sufficiently rare case that we don't have to worry about.)

I have gone ahead and done that in e315e9b.

I did have a look at your images workflow and i am not feeling really confident about implementing it correctly on the first try.

Given that this PR is holding up SOFTWARE-5154 (the image is not built and thus not available from harbor), i am thinking just to get this PR merged.  Then, if you still want your robustness improvement, and you think that putting `docker.io` first in the list won't do just that, then I can put that in a separate PR as not to hold up the show for SOFTWARE-5154.

What do you think?

Comment (brianhlin) [reply .github/workflows/build-container.yml:68]: Sounds reasonable to me!

**Pull request merged**